### PR TITLE
Text field new UI

### PR DESCRIPTION
**Pull request by kjkurtz**
### **PR Type**
Enhancement, Tests

___

### **Description**
- Enhanced `TextField` component with new props (`title`, `state`, `helperText`, `errorText`, `placeholderText`) and improved UI.
- Updated various components (`Field`, `WebAddressAutocomplete`, `UnifiedAddressAutoComplete`, `MobileAddressAutoComplete`, `CustomSelect`) to align with new `TextField` props.
- Updated storybook demos and configurations to showcase new `TextField` features and states.
- Removed deprecated props and components for cleaner codebase.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><details><summary>7 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>TextField.tsx</strong><dd><code>Enhanced TextField component with new props and improved UI.</code></dd></summary>
<hr>
      
packages/ui/src/TextField.tsx

<li>Added new props: <code>title</code>, <code>state</code>, <code>helperText</code>, <code>errorText</code>, <code>placeholderText</code>.<br> <li> Updated styles and layout for better UI consistency.<br> <li> Removed <code>WithLabel</code> component usage.<br> <li> Refactored conditional rendering and event handling.<br>

</details>
    

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/601/files#diff-526ca8a54e7f6aaa340ccd6be7bb641b6ee65087e2455eef8ac1dad81625c433">+165/-185</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Field.tsx</strong><dd><code>Updated Field component to align with new TextField props.</code></dd></summary>
<hr>
      
packages/ui/src/Field.tsx

<li>Updated <code>Field</code> component to use new <code>TextField</code> props.<br> <li> Replaced <code>disabled</code> prop with <code>state</code> prop.<br> <li> Adjusted placeholder prop to <code>placeholderText</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/601/files#diff-b179881630547b40643ddd11ccdc12624a62bf6deafcea924246db78416d25d5">+18/-18</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Common.ts</strong><dd><code>Updated Common interfaces for TextField enhancements.</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
packages/ui/src/Common.ts

<li>Added <code>TextStyleWithOutline</code> interface.<br> <li> Updated <code>TextFieldProps</code> interface with new props.<br> <li> Removed deprecated props.<br>

</details>
    

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/601/files#diff-538e93b553971318d69ff6fff2ac103f295c4922e24bbe61c30897baa463bba1">+14/-13</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>WebAddressAutocomplete.tsx</strong><dd><code>Updated WebAddressAutocomplete to align with new TextField props.</code></dd></summary>
<hr>
      
packages/ui/src/WebAddressAutocomplete.tsx

<li>Updated to use new <code>TextField</code> props.<br> <li> Replaced <code>disabled</code> prop with <code>state</code> prop.<br> <li> Adjusted placeholder prop to <code>placeholderText</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/601/files#diff-111c1c6aafbe3792f55df006a2c73279c4f6cd32a6858a925dde6c7610f2fdf5">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>UnifiedAddressAutoComplete.tsx</strong><dd><code>Updated UnifiedAddressAutoComplete to align with new TextField props.</code></dd></summary>
<hr>
      
packages/ui/src/UnifiedAddressAutoComplete.tsx

<li>Updated to use new <code>TextField</code> props.<br> <li> Replaced <code>disabled</code> prop with <code>state</code> prop.<br> <li> Adjusted placeholder prop to <code>placeholderText</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/601/files#diff-37ed91dd74d62270efebb005d85860588a4cc2ea7259a7f62adf6ca9c0667b50">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>MobileAddressAutoComplete.tsx</strong><dd><code>Updated MobileAddressAutoComplete to align with new TextField props.</code></dd></summary>
<hr>
      
packages/ui/src/MobileAddressAutoComplete.tsx

<li>Updated to use new <code>TextField</code> props.<br> <li> Replaced <code>disabled</code> prop with <code>state</code> prop.<br> <li> Adjusted placeholder prop to <code>placeholderText</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/601/files#diff-0023dd2015e74f101fca29a3b63dd7d922e63b734a59f9a4682d8ed99fd57f27">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>CustomSelect.tsx</strong><dd><code>Updated CustomSelect to align with new TextField props.</code>&nbsp; &nbsp; </dd></summary>
<hr>
      
packages/ui/src/CustomSelect.tsx

<li>Updated to use new <code>TextField</code> props.<br> <li> Replaced <code>disabled</code> prop with <code>state</code> prop.<br> <li> Adjusted placeholder prop to <code>placeholderText</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/601/files#diff-fdeffd12418e66a49aaf6dd3fa7619008191cbe6dca0cad9d28148b8cbec8571">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Tests
</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>TextField.stories.tsx</strong><dd><code>Updated TextField storybook demos with new props and states.</code></dd></summary>
<hr>
      
apps/demo/stories/TextField.stories.tsx

<li>Updated storybook demos for <code>TextField</code> component.<br> <li> Added new demos for different states and props.<br> <li> Removed old story implementations.<br>

</details>
    

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/601/files#diff-f13cab81561d64d09d5ea0388899d02a1a954eab15731a48c5c1def86f193914">+49/-66</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>TextField.config.tsx</strong><dd><code>Updated demo configuration for new TextField demos.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
apps/demo/story-config/TextField.config.tsx

<li>Updated demo configuration to use new TextField demos.<br> <li> Added new demo entries for different TextField states.<br>

</details>
    

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/601/files#diff-76387cdc1877e2612643adae7bd830f671b2ff83f27e6879d8bf1b977606f3ba">+23/-7</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: packages/ui/src/TextField.tsx [hunk 1]
```diff
@@ -241,121 +215,140 @@ export const TextField = ({
...
+            placeholder={placeholderText}
+            placeholderTextColor={theme.text.secondaryLight}
+            returnKeyType={type === "number" || type === "decimal" ? "done" : returnKeyType}
+            secureTextEntry={type === "password"}
+            style={defaultTextInputStyles}
+            testID={testID}
+            textContentType={textContentType}
+            underlineColorAndroid="transparent"
+            value={displayValue}
+            onBlur={() => {
```
Comment (hchamorro) [inline packages/ui/src/TextField.tsx:291]: will you please add prop `selectable` to TextInput
either 
```
const isSelectable = !disabled
  selectable={isSelectable}
// or
selectable= !disabled
```

This issue-  The user should not be able to select if disabled, we do not want onBlur function to run if element is disabled.

**Review by hchamorro — COMMENTED** [review]
Looks Great!

Open to  conversations about adding state as a prop. 

I think the disabled prop can be passed and state should be handled within the component.
(That is, unless the intention it the state is handled by field and prop drilled to child)

I do strongly feel that we should add an onError prop.
It is a function
if boolean than update to error state internally

File: packages/ui/src/TextField.tsx (hunk 1 above)
Comment (kjkurtz) [reply packages/ui/src/TextField.tsx:291]: Is there a way to simply disable selection on `TextInput` that you know of? That is a react-native component and I didn't see an easy way without manually changing borders, selection characteristics, etc. It is an issue with our current `TextField` as well.

**Comment by kjkurtz** [comment]
- Switched back to having a `disabled` prop and using the existence of `errorText` for determining if the error state should be shown. 
- I did not add onError at this point as I think we need to figure out what the use case is there, and would be an easy add on at a later time.
- I could not find an easy way to change the style on disabled of TextInput without getting deep into CSS world (and its an existing problem at least for web) so let that be for now.

**Pull request merged**